render-test/runner.cpp

```diff
@@ -1035,14 +1035,16 @@ bool TestRunner::runOperations(const std::string& key, TestMetadata& metadata, R
 
         map.flyTo(mbgl::CameraOptions().withCenter(endPos).withZoom(endZoom), animationOptions);
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wfor-loop-analysis"
```

Comment:
I was also fixing this, but I converted it to a `while`. Can we do the same here?
https://github.com/mapbox/mapbox-gl-native/pull/15890/files#diff-1e3ea4c400855d268ae2335be0bbc0d9

Reply:
It is "compiler golf", but I think it is cleaner them a pragma

Reply:
👍 Thanks for the suggestion, def cleaner to avoid the pragma: done in 96488ad